import "VisionObservations.proto";
import "WorldInfo.proto";


message ActionDescription{
	enum URGENCY_LEVELS{
		SOMEDAY=0;
		SLOW=1;
		BEAM_ME_UP=2;
	}
		
	required uint32 action=1 [default = 0];
	required uint32 urgency=2 [default = 0]; 
	required string description=3 [default = ""];
	required float x=4 [default = -1];
	required float y=5 [default = -1];
	required float phi=6 [default = -1];
	required float speed=7 [default = -1];
}
message BodyControlMessage{
	enum BODYACTIONS{
		NOTHING=0;
		STAND=1;
		DOACTION=2;
		BALLAPPROACH=3; //Target (=orientation) the opposing goal post
		BALLAPPROACH_SUPPORT=4; //Target the side of the goal post
		GOTO=5;  //Use x,y, phi to describe where
		DEMO=6; //Reserved
	}
	required ActionDescription task=1;
}
message HeadControlMessage{
	enum HEADACTIONS{
		NOTHING=0;
		FROWN=1; //Look down
		SCAN_AND_TRACK_FOR_BALL=2;
		LOCALIZE=3;
		LOCALIZE_FAR=4;
		SMART_SELECT=5; //Decide based on the distance from the ball, and the urgency
		
	}
	required ActionDescription task=1;
}
message BallFoundMessage{
	required bool ballfound=1;
}
